### ISSUE TITLE
examples/array_backed_grid.py: TypeError: on_mouse_scroll() takes 3 positional arguments but 5 were given

### ISSUE THREAD

**Issue by iacopy** [pvcraven/arcade#143]
It happens when clicking on an already green cell.
```
$ python array_backed_grid.py
Click coordinates: (80, 447). Grid coordinates: (12, 2)
Click coordinates: (188, 359). Grid coordinates: (10, 5)
Click coordinates: (159, 377). Grid coordinates: (10, 4)
Click coordinates: (199, 418). Grid coordinates: (11, 5)
Click coordinates: (224, 361). Grid coordinates: (10, 6)
Traceback (most recent call last):
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/event.py", line 367, in dispatch_event
    if getattr(self, event_type)(*args):
TypeError: on_mouse_scroll() takes 3 positional arguments but 5 were given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 234, in 'calling callback function'
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 1121, in objc_method
    result = f(py_self, *args)
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/window/cocoa/pyglet_view.py", line 239, in scrollWheel_
    self._window.dispatch_event('on_mouse_scroll', x, y, scroll_x, scroll_y)
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/window/__init__.py", line 1220, in dispatch_event
    if EventDispatcher.dispatch_event(self, *args) != False:
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/event.py", line 371, in dispatch_event
    event_type, args, getattr(self, event_type))
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/event.py", line 392, in _raise_dispatch_exception
    inspect.getargspec(handler)
  File "/Users/iacopomarmorini/.pyenv/versions/3.6.3/lib/python3.6/inspect.py", line 1072, in getargspec
    raise ValueError("Function has keyword-only parameters or annotations"
ValueError: Function has keyword-only parameters or annotations, use getfullargspec() API which can support them
Traceback (most recent call last):
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/event.py", line 367, in dispatch_event
    if getattr(self, event_type)(*args):
TypeError: on_mouse_scroll() takes 3 positional arguments but 5 were given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 234, in 'calling callback function'
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 1121, in objc_method
    result = f(py_self, *args)
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/window/cocoa/pyglet_view.py", line 239, in scrollWheel_
    self._window.dispatch_event('on_mouse_scroll', x, y, scroll_x, scroll_y)
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/window/__init__.py", line 1220, in dispatch_event
    if EventDispatcher.dispatch_event(self, *args) != False:
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/event.py", line 371, in dispatch_event
    event_type, args, getattr(self, event_type))
  File "/Users/iacopomarmorini/.virtualenvs/arcade/lib/python3.6/site-packages/pyglet/event.py", line 392, in _raise_dispatch_exception
    inspect.getargspec(handler)
  File "/Users/iacopomarmorini/.pyenv/versions/3.6.3/lib/python3.6/inspect.py", line 1072, in getargspec
    raise ValueError("Function has keyword-only parameters or annotations"
ValueError: Function has keyword-only parameters or annotations, use getfullargspec() API which can support them
Click coordinates: (224, 361). Grid coordinates: (10, 6)
```

**Comment by pvcraven** [comment]
Already fixed in the 1.2.2 branch with this checkin: https://github.com/pvcraven/arcade/commit/b207a469d48192b81b078768ab8f9b111334476f

**Comment by pvcraven** [comment]
Closed, implemented in 1.2.2 which is now deployed on PyPi.
